### DISCUSSION TITLE
Milvus doesn't work on hyper-v. it works on virtual box, qemu kvm, but not on hyprt-v

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#23362] [Q&A and General discussion]
How do I make milvus work on hyper-v? 
Have anyone succeeded in it?

**Comment by yhmo** [comment]
We didn't try it on hyper-v.
Are you running it on docker?

**Reply by martinenkoEduard** [reply]
I tried it with .deb and docker compose. In botch cases it fails to start on Hyper-V.

**Reply by yhmo** [reply]
What is the failed error in the log?

**Reply by martinenkoEduard** [reply]
Something with etcd.
I will try to replicate the error and upload the logs.

**Reply by DanZai233** [reply]
> Something with etcd. I will try to replicate the error and upload the logs.

I have the same problem but it only happends on milvus version 2.2.x
I think it is the incompatible version of etcd and milvus 2.2.x

**Reply by martinenkoEduard** [reply]
Yep. It is a problem with ETCD. Don't know why it happens on hyper-v only.
